### PR TITLE
Affine substitution Cipher - C++

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented Affine substitution Cipher in C++.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#486

### CODE REVIEW

File: C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+const int a = 19;
+const int b = 20;
+const int m = 26;
+
+string encryption(string message) {
+    string cipher;
+    for (int i = 0; i < message.size(); ++i) {
+        if(message[i]!=' ') {
+            cipher += (char)((a * (message[i] - 'A') + b) % 26 + 'A');
```
Comment (iamrajiv) [inline C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp]: @SangeetaMishra make a comment here i.e. ` to make it in range of ASCII alphabet addition of "A" is done`

File: C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    int a_inv = 0;
+    for (int i = 0; i < 26; ++i) {
+        if((a * i) % 26 == 1){
+            a_inv = i;
+            break;
+        }
+    }
+    for (int i = 0; i < cipher.size(); ++i) {
+        if(cipher[i] != ' '){
+            message += (char)(a_inv * (cipher[i] + 'A' - b) % 26 + 'A');
```
Comment (iamrajiv) [inline C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp]: @SangeetaMishra make the same comment here too

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SangeetaMishra I would also like to suggest of adding affine functions used in the program in the code comments and also try to make it more informative like clarifying why space is used int `if(message[i]!=' ')` one reason maybe like `to avoid spaces to be encrypted or decrypted`

File: C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        else {
+            message += cipher[i];
+        }
+    }
+    return message;
+}
+
+int main() {
+    string message = "TESSERACTCODINGNEOALGO";
+    //use only uppercase characters (you can modift the code for lowercase characters)
```
Comment (atarax665) [inline C-Plus-Plus/cryptography/Affine_substitution_Cipher.cpp]:
```suggestion
    //use only uppercase characters (you can modify the code for lowercase characters)
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]
Great! :+1:

**Review by siddharth25pandey — APPROVED** [review]
@SangeetaMishra Good to go
Able to merge.
Thanks!

**Pull request merged**